Frontend/js/raytracing.js

```diff
@@ -5,6 +5,99 @@ var arrayListForObject 		 = [];
 
 var globalImagePlaneSizeObject, globalSceneObject, globalRaytracerObject, globalAmbientLight;
 
+// Arrays of information for shapes in the scene
+var responsive_shpe_number = 0;
+var responsive_shpe_name = [];
+var responsive_shpe_x = [];
+var responsive_shpe_y = [];
+var responsive_shpe_z = [];
+var responsive_shpe_size = [];
+var tmp_responsive_shpe_size = 0;

```

Comment:
Why are you using tmp_names? do they mean something or did you forget to change them? It'll be great if you can rename it into something which conveys more meaning. Thanks!